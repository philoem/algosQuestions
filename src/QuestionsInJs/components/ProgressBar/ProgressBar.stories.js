import ProgressBar from './index'

export default {
	component: ProgressBar,
	title: 'Components/ProgressBar'
}

const Template = (args) => <ProgressBar {...args} />

export const Default = Template.bind({})
