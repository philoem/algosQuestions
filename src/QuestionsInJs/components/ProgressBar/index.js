import React from 'react'
import progressBar from '../../../utils/progressBar'
import styles from '../../CardsQuestions/index.module.css'
import PropTypes from 'prop-types'

const ProgressBar = ({ questionLength, index }) => {
	return (
		<div className={styles.progressContainer}>
			<div
				className={styles.progress}
				style={{
					width: `${progressBar({ questionLength, index })}%`,
					backgroundColor: '#2A265F'
				}}
				// eslint-disable-next-line jsx-a11y/aria-role
				role='progressBar'
			/>
			<span className={styles.progressText}>
				{index + 1}/{questionLength} Questions
			</span>
		</div>
	)
}

ProgressBar.defaultProps = {
	questionLength: null,
	index: 0
}

ProgressBar.propTypes = {
	questionLength: PropTypes.number,
	index: PropTypes.number
}

export default ProgressBar
