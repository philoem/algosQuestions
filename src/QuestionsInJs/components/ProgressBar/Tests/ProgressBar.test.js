import React from 'react'
import { render, screen } from '@testing-library/react'
import { defineFeature, loadFeature } from 'jest-cucumber'
import ProgressBar from '../../ProgressBar/index'

const feature = loadFeature('src/QuestionsInJs/components/ProgressBar/Tests/ProgressBar.feature')

defineFeature(feature, (test) => {
	let container

	const givenProgressBarComponentIsRendered = (given, comp) => {
		given('ProgressBar component is rendered', () => {
			container = render(comp).container
		})
	}

	const andRequiredElementsAreMounted = (and) => {
		and('Required elements are mounted', () => {
			expect(screen.getByRole('progressBar')).toBeInTheDocument()
			expect(screen.getByText(/Questions/i)).toBeInTheDocument()
		})
	}

	const andDefaultStyleIsApplied = (and) => {
		and('Default style is applied', () => {
			expect(container).toMatchSnapshot()
		})
	}

	test('Default progress bar', ({ given, and }) => {
		givenProgressBarComponentIsRendered(given, <ProgressBar questionLength={6} index={0} />)
		andRequiredElementsAreMounted(and)
		andDefaultStyleIsApplied(and)

		and('Correct value is applied', () => {
			expect(screen.getByRole('progressBar')).toHaveStyle('width: 17%')
			expect(screen.getByRole('progressBar')).toHaveStyle('backgroundColor: #2A265F')
		})
	})

	test('Math round value', ({ given, and }) => {
		givenProgressBarComponentIsRendered(given, <ProgressBar questionLength={6} index={3} />)
		andRequiredElementsAreMounted(and)
		andDefaultStyleIsApplied(and)

		and('Value is rounded', () => {
			expect(screen.getByRole('progressBar')).toHaveStyle('width: 67%')
		})
	})
})
