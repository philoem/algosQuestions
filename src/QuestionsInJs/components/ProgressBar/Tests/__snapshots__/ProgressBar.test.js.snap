// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Default progress bar 1`] = `
<div>
  <div
    class="progressContainer"
  >
    <div
      class="progress"
      role="progressBar"
      style="width: 17%; background-color: rgb(42, 38, 95);"
    />
    <span
      class="progressText"
    >
      1
      /
      6
       Questions
    </span>
  </div>
</div>
`;

exports[`ProgressBar Math round value 1`] = `
<div>
  <div
    class="progressContainer"
  >
    <div
      class="progress"
      role="progressBar"
      style="width: 67%; background-color: rgb(42, 38, 95);"
    />
    <span
      class="progressText"
    >
      4
      /
      6
       Questions
    </span>
  </div>
</div>
`;
